### DISCUSSION TITLE
V 3.9.1: not all compilers support exceptions

### DISCUSSION THREAD

**Discussion by SDziamchuk** [nlohmann/json#2479] [Q&A] [answered]
Hello! I'm use this library in my mbed-os projects. But in last update exceptions was added. And ARM Compiler does not support exceptions:
```
[Error] json3.9.1.hpp@16457,9: cannot use 'throw' with exceptions disabled
[Error] json3.9.1.hpp@16470,9: cannot use 'throw' with exceptions disabled
```
ARM Compiler version: 6.14.
Is it possible not to use exceptions for more compatibility with different compilers?

**Comment by nlohmann** [comment]
Yes, they can be switched off. See

- https://nlohmann.github.io/json/home/exceptions/#switch-off-exceptions
- https://github.com/nlohmann/json#further-notes (last item)

**Reply by SDziamchuk** [reply]
According to this https://os.mbed.com/docs/mbed-os/v6.4/program-setup/build-profiles-and-rules.html
flag `-fno-exceptions` is set by default for GCC_ARM.
I added define to my code
```
#define JSON_THROW_USER(exception)  {std::abort();}
#include "../third_party/nlohmann/json.hpp"
```
But nothing happened. I got same error.

**Reply by nlohmann** [reply]
Can you try just defining `JSON_NOEXCEPTION`?

**Reply by SDziamchuk** [reply]
Yes, I tried. It does not help.

**Comment by gregmarr** [comment, marked as answer]
@nlohmann Those two lines do have raw `throw` calls.

    throw std::out_of_range("key not found");

**Reply by gregmarr** [reply]
Fixed in this PR https://github.com/nlohmann/json/pull/2347

**Reply by gregmarr** [reply]
@SDziamchuk This should be fixed in the latest code.
